_data-prepper/pipelines/configuration/processors/mutate-event.md

```diff
@@ -0,0 +1,306 @@
+* [ConvertEntry] (#ConvertEntry)
+
+--->
+
+## AddEntries
+
+The `AddEntries` processor adds entries to an event.
+
+### Configuration
+
+You can configure the `AddEntries` processor with the following options.
+
+Option | Required | Description
+:--- | :--- | :---
+|`entries` | Yes | A list of entries to add to an event. |
+|`key` | Yes | The key of the new entry to be added. |
+|`value` | Yes | The value of the new entry to be added. You can use the following data types: strings, booleans, numbers, null, nested objects, and arrays. |
+|`overwrite_if_key_exists` | No | When set to `true`, if `key` already exists in the event, then the existing value is overwritten. The default value is `false`. |
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - add_entries:
+        entries:
+        - key: "newMessage"
+          value: 3
+          overwrite_if_key_exists: true
+  sink:
+    - stdout:
+```
+
+Create a log file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with this filepath.
+
+```json
+{"message": "value"}
+```
+
+When you run this processor, it parses the message into the following output:
+
+```json
+{"message": "value", "newMessage": 3}
+```
+
+> If `newMessage` already exists, its existing value is overwritten with a value of `3`.
+
+## copy value
+
+The `copy value` processor copies values within an event.
+
+### Configuration
+
+You can configure the `copy value` processor with the following options.
+
+Option | Required | Description
+:--- | :--- | :---
+| `entries` | Yes | A list of entries to be copied in an event. |
+| `from_key` | Yes | The key of the entry to be copied.
+| `to_key` | Yes | The key of the new entry to be added.
+| `overwrite_if_to_key_exists` | No | When set to a value of `true`, if `to_key` already exists in the event, then the existing value will be overwritten. The default value is `false`. |
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - copy_values:
+        entries:
+        - from_key: "message"
+          to_key: "newMessage"
+          overwrite_if_to_key_exists: true
+  sink:
+    - stdout:
+```
+
+Create a log file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file:
+
+```json
+{"message": "value"}
+```
+
+When you run this processor, it parses the message into the following output:
+
+```json
+{"message": "value", "newMessage": "value"}
+```
+
+> If `newMessage` had already exists, its current value is overwritten with `value`.
+
+
+## DeleteEntry
+
+The `DeleteEntry` processor deletes entries in an event.
```

Comment:
@oeyh Which entries? How do you define what you want to delete?

Reply:
Entries in the event, i.e. key-value pairs. You define the keys to delete in the `with_keys` field under `delete_entries` in the yaml file, and those keys along with their values (i.e. those entries) will be deleted.